app/models/remote_theme.rb

```diff
@@ -457,6 +459,50 @@ def update_theme_color_schemes(theme, schemes)
     theme.color_scheme = ordered_schemes.first if theme.new_record?
   end
 
+  def upsert_theme_site_settings(theme, theme_site_settings)
```

Comment:
I guess potentially I could move this code into a service call, but then I would need to call service from a service, so not sure